src/main/kotlin/com/autonomousapps/tasks/DominatorViewTask.kt

```diff
@@ -0,0 +1,159 @@
+@file:Suppress("SpellCheckingInspection")
+
+package com.autonomousapps.tasks
+
+import com.autonomousapps.TASK_GROUP_DEP
+import com.autonomousapps.internal.graph.DominanceTree
+import com.autonomousapps.internal.graph.GraphWriter
+import com.autonomousapps.internal.utils.FileUtils
+import com.autonomousapps.internal.utils.FileUtils.byteCountToDisplaySize
+import com.autonomousapps.internal.utils.fromJson
+import com.autonomousapps.internal.utils.fromJsonSet
+import com.autonomousapps.internal.utils.getAndDelete
+import com.autonomousapps.model.Coordinates
+import com.autonomousapps.model.DependencyGraphView
+import com.autonomousapps.model.PhysicalArtifact
+import com.autonomousapps.model.ProjectCoordinates
+import com.google.common.graph.Graph
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.support.appendReproducibleNewLine
+import java.io.File
+
+@Suppress("UnstableApiUsage") // Guava graphs
+abstract class DominatorViewTask : DefaultTask() {
+
+  init {
+    group = TASK_GROUP_DEP
+    description = "Prints a dominator view of the dependency graph"
+
+    // TODO temporary until task split into output-creating and console-printing
+    outputs.upToDateWhen { false }
+  }
+
+  @get:Input
+  abstract val projectPath: Property<String>
+
+  @get:PathSensitive(PathSensitivity.NONE)
+  @get:InputFile
+  abstract val physicalArtifacts: RegularFileProperty
```

Comment:
Should it rather be `@Input Configuration`?

Reply:
Using this input lets me re-use existing plugin infrastructure. To achieve your desired goal of printing arbitrary configurations, this would indeed have to change. I need to take some time to consider how best to implement it for the various use-cases that might be worth supporting, including Android as well as JVM.

It's very frustrating that Gradle tasks only support flags when you invoke the specific task that takes the flag. It would be great if a flag (like `--configuration runtimeClasspath`) could be consumed my multiple tasks in the task graph. Because that's not the case, I need to explode the size of this relatively simple task and have it duplicate functionality from other tasks. I don't want to do that without careful consideration. I don't want to create headaches for my future self.